source/index-management/index-management.rst

```diff
@@ -852,6 +852,181 @@ Example::
+until index changes appear.
+
+An example, from Javascript:
+
+.. code:: typescript
+
+  const name = await collection.createSearchIndex({ definition: { ... fill out definition } })
+  while (!(await collection.listSearchIndexes({ name }).hasNext())) {
+    await setTimeout(1000);
+  }
+ 
+Common Interfaces
+-----------------
+
+.. code:: typescript
+
+  interface SearchIndexModel {
+    // Returns the definition for this index.
+    definition: Document;
+
+    // Returns the name for this index, if present.
+    name: Optional<string>;
+  }
+
+  /**
+   * The following interfaces are empty but are provided as placeholders for drivers that cannot 
+   * add options in a non-breaking manner, if options are added in the future.
+   */
+  interface CreateSearchIndexOptions {} 
+  interface UpdateSearchIndexOptions {}
+  interface ListSearchIndexOptions {}
+  interface DropSearchIndexOptions {}
+
+Standard API for Search Indexes
+-------------------------------
+
+.. code:: typescript
+
+  interface Collection {
+    /**
+    * Convenience method for creating a single search index.
+    * 
+    * @return The name of the created search index
+    * @note Drivers MAY opt to implement this method signature, the signature that
+    *   takes an SearchIndexModel as a parameter, or for those languages with method
+    *   overloading MAY decide to implement both.
+    */
+    createSearchIndex(name: String, definition: Document, options: Optional<CreateSearchIndexOptions>): String;
+
+    /**
+    * Convenience method for creating a single index.
+    * 
+    * @return The name of the created search index
+    * 
+    * @note Drivers MAY opt to implement this method signature, the signature that
+    *   takes an name and a definition as parameters, or for those languages with method
+    *   overloading MAY decide to implement both.
+    */
+    createSearchIndex(model: SearchIndexModel, options: Optional<CreateSearchIndexOptions>): String;
+
+    /**
+    * Creates multiple search indexes on the collection.
+    * 
+    * @return An iterable of the newly created index names.
+    */
+    createSearchIndexes(models: Iterable<SearchIndexModel>, options: CreateSearchIndexOptions): Iterable<String> ;
+
+    /**
+    * Updates the search index with the given name to use the provided 
+    * definition.
+    */
+    updateSearchIndex(name: String, definition: Document, options: Optional<UpdateSearchIndexOptions>): void;
+
+    /**
+    * Drops the search index with the given name.
+    */
+    dropSearchIndex(name: String, options: Optional<DropSearchIndexOptions>): void;
+
+    /**
+    * Gets index information for all search indexes in the collection.
+    */
+    listSearchIndexes(name: Optional<String>, aggregationOptions: Optional<AggregationOptions>, listIndexOptions: Optional<ListSearchIndexOptions>): Cursor<{ name: String }>;
```

Comment:
Are we not allowing `id` to be specified?

Reply:
Correct, you can see this slack thread for context: https://mongodb.slack.com/archives/C04TMC263M3/p1683837096346969?thread_ts=1683836146.256049&cid=C04TMC263M3